tree/30_generic_methods/service_stopped.cf

```diff
@@ -55,11 +55,13 @@ bundle agent service_stopped(name)
       "${report_data.method_id}_${name}" usebundle => ncf_services("${name}", "stop");
       # Use stop result as method result
       "${report_data.method_id}_${name}" usebundle => call_method_classes("${class_prefix}");
+      "${report_data.method_id}_${name}" usebundle => call_method_classes("${report_data.report_id}");
       "${report_data.method_id}_${name}" usebundle => call_method_classes_caller;
       "${report_data.method_id}_${name}" usebundle => call_method_end("ncf_services_stop");
     pass3.!needs_stop::
       # Define success classes is check_running was not kept
       "${report_data.method_id}_${name}" usebundle => _classes_success("${class_prefix}");
+      "${report_data.method_id}_${name}" usebundle => _classes_success("${report_data.report_id}");
```

Comment:
looks good, but I feel like we will have issues sooner or later with newer methods calling non-leaf methods that are defining classes on the report_id. If we handle correctly the push pop process, should'nt we be able to always define classes on the method_id and only use the report_id variable as a match for reporting?